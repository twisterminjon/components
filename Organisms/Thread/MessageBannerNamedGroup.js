import React from 'react'
import PropTypes from 'prop-types'
import { Popup } from 'semantic-ui-react'

import AvatarNamedGroup from './AvatarNamedGroup'
import IconAddUser from '../../Atoms/Icons/IconAddUser'

import './MessageBanner.css'

MessageBanner.propTypes = {
  /** If true, will show add participant button */
  canModifyMembers: PropTypes.bool,

  /** Name of the group */
  groupName: PropTypes.string.isRequired,

  /** List of group members */
  members: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      displayName: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,

  /** function called after the add/edit action */
  onAdd: PropTypes.func.isRequired,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}

MessageBanner.defaultProps = {
  canModifyMembers: true,
  className: '',
  style: {},
}

export default function MessageBanner({ canModifyMembers, groupName, members, onAdd, className, style }) {
  const handleAdd = () => {
    onAdd({ editing: true })
  }

  const memberText = `${members.length} Members`
  return (
    <React.Fragment>
      <div className={`messagebanner ${className}`} style={style} data-testid="message-banner">
        <div className="messagebanner-avatar-wrap">
          <AvatarNamedGroup />
        </div>
        <Popup
          inverted
          hoverable
          position="bottom center"
          className="messagebannernamedgroup-member-wrap"
          trigger={
            <div className="messagebannernamedgroup-text-content">
              <span>{groupName}</span>
              <span className="messagebanner-label">{memberText}</span>
            </div>
          }>
          <MemberList members={members} />
        </Popup>
        {canModifyMembers && (
          <button
            className="messagebanner-button messagebanner-adduser-button"
            onClick={handleAdd}
            data-testid="edit-group">
            <IconAddUser color={'var(--button_primary_bg)'} size="34" />
          </button>
        )}
      </div>
    </React.Fragment>
  )
}

const MemberList = ({ members }) => {
  return (
    <div className="messagebannernamedgroup-member-list">
      <h6>{members.length} Group Members</h6>
      {members.map(user => (
        <span key={user.id}>{user.displayName}</span>
      ))}
    </div>
  )
}
