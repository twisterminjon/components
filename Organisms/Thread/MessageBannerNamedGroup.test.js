import React from 'react'

import MessageBannerNamedGroup from './MessageBannerNamedGroup'
import users from '../../../Mocks/Users.mock'

describe('MessageBannerNamedGroup', () => {
  const mockFun = jest.fn()

  it('matches the snapshot to allow calling', () => {
    const wrapper = window.shallow(<MessageBannerNamedGroup groupName="test" members={users} onAdd={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })
})
