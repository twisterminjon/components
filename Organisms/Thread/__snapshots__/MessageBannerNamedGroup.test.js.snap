// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBannerNamedGroup matches the snapshot to allow calling 1`] = `
<Fragment>
  <div
    className="messagebanner "
    data-testid="message-banner"
    style={Object {}}
  >
    <div
      className="messagebanner-avatar-wrap"
    >
      <NamedGroupAvatar
        className=""
        style={Object {}}
      />
    </div>
    <Popup
      className="messagebannernamedgroup-member-wrap"
      disabled={false}
      eventsEnabled={true}
      hoverable={true}
      inverted={true}
      on={
        Array [
          "click",
          "hover",
        ]
      }
      pinned={false}
      popperModifiers={Array []}
      position="bottom center"
      trigger={
        <div
          className="messagebannernamedgroup-text-content"
        >
          <span>
            test
          </span>
          <span
            className="messagebanner-label"
          >
            8 Members
          </span>
        </div>
      }
    >
      <MemberList
        members={
          Array [
            Object {
              "displayName": "Agnes Rogers, MD",
              "id": "3",
              "overallStatus": "available",
              "profileImage": "https://www.fillmurray.com/200/200",
            },
            Object {
              "displayName": "David McGinnis, MD",
              "id": "9",
              "overallStatus": "offline",
              "profileImage": "https://www.fillmurray.com/400/400",
            },
            Object {
              "displayName": "David Smith, MD",
              "id": "4",
              "overallStatus": "available",
              "profileImage": "https://www.fillmurray.com/600/600",
            },
            Object {
              "displayName": "Jennifer Reyes",
              "id": "8",
              "overallStatus": "away",
              "profileImage": "https://www.fillmurray.com/500/500",
            },
            Object {
              "displayName": "Paul Jason, RN",
              "id": "6",
              "overallStatus": "busy",
              "profileImage": "https://www.fillmurray.com/700/700",
            },
            Object {
              "displayName": "Rosalyn Stratton, RN",
              "id": "5",
              "overallStatus": "offline",
              "profileImage": "https://www.fillmurray.com/800/800",
            },
            Object {
              "displayName": "Sherry Jackson, MD",
              "id": "10",
              "overallStatus": "offline",
              "profileImage": "https://www.fillmurray.com/200/200",
            },
            Object {
              "displayName": "Tony Hunter",
              "id": "7",
              "overallStatus": "available",
              "profileImage": "https://www.fillmurray.com/450/450",
            },
          ]
        }
      />
    </Popup>
    <button
      className="messagebanner-button messagebanner-adduser-button"
      data-testid="edit-group"
      onClick={[Function]}
    >
      <IconAddUser
        className=""
        color="var(--button_primary_bg)"
        size="34"
        style={Object {}}
      />
    </button>
  </div>
</Fragment>
`;
